### ISSUE TITLE
ray.wait with local_mode=True blocks for a very long time

### ISSUE THREAD

**Issue by sbuschjaeger** [ray-project/ray#7741]
Consider the following code:

    @ray.remote
    def start_experiment(param):
        # do some stuff
       print("DONE")
       return experiment_results

    def start_experiments(parameters):
        ray.init(address=None, local_mode=True)
        futures = [start_experiment(param) for param in parameters]
        while futures:
            result, futures = ray.wait(futures)
            print("WAIT IS READY")
            eval_return = ray.get(result[0])
            print("GET IS READY")

            out_file_content = eval_return
            with open("/results.jsonl", "a", 1) as out_file:
                out_file.write(out_file_content)

    
For debugging purposes I am currently running ray with local_mode=True. The experiments are very small and take about 1 min each. "DONE" gets immediately printed once the experiment is over. The data involved is only a few Megabytes. In this scenario wait() ~never~ unblocks after a very long time (~10 Min+), even though all experiments are already done. 

#499 mentions a similar problem, but is closed (and seems fixed?)
#5060 also goes into some details, but a fix is already merged (not sure though?)

I am using ray version 0.8.2.

**Comment by rkooo567** [comment]
@ijrsvt

**Comment by ijrsvt** [comment]
@sbuschjaeger does this run on the current version of ray?

**Comment by sbuschjaeger** [comment]
Sorry for the late reply. Updated to ray 0.8.5, but the problem remains

**Comment by rkooo567** [comment]
This must be a bug in this case? @ijrsvt again

**Comment by cszhu** [comment]
Hello! This P2 issue has seen no activity in at least 2 years. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
